src/sinks/appsignal/mod.rs

```diff
@@ -0,0 +1,268 @@
+use std::io::Write;
+
+use bytes::{BufMut, Bytes, BytesMut};
+use flate2::write::GzEncoder;
+use futures::{FutureExt, SinkExt};
+use http::{header::AUTHORIZATION, Request, Uri};
+use hyper::Body;
+use serde_json::json;
+use vector_common::sensitive_string::SensitiveString;
+use vector_config::configurable_component;
+
+use crate::{
+    codecs::Transformer,
+    config::{AcknowledgementsConfig, DataType, Input, SinkConfig, SinkContext},
+    event::Event,
+    http::HttpClient,
+    sinks::{
+        util::{
+            http::{BatchedHttpSink, HttpEventEncoder, HttpSink},
+            BatchConfig, BoxedRawValue, JsonArrayBuffer, SinkBatchSettings, TowerRequestConfig,
+        },
+        BuildError,
+    },
+};
+
+/// Configuration for the `appsignal` sink.
+#[configurable_component(sink("appsignal"))]
+#[derive(Clone, Debug, Default)]
+pub struct AppsignalSinkConfig {
+    /// The AppSignal API endpoint to report to. This is configured by default and doesn't need to be changed.
+    #[configurable(validation(format = "uri"))]
+    #[configurable(metadata(docs::examples = "https://appsignal-endpoint.net"))]
+    endpoint: Option<String>,
```

Comment:
```suggestion
    #[serde(default = "https://appsignal-endpoint.net"))]
    endpoint: String,
```

This will produce a better rendered user docs.

Reply:
I couldn't specify a String value directly, it expected a function I think. As described in the [Serde docs for defaults](https://serde.rs/attr-default.html).

Reply:
Doh, I always forget if it's a function with the `()` removed or not. I think our configurable trait uses a function when the `()` is included 🙃